code/game/objects/items/weapons/lighters.dm

```diff
@@ -18,11 +18,14 @@
 	var/next_on_message
 	/// Cooldown until the next turned off message/sound can be activated
 	var/next_off_message
+	/// Our lighter color suffix. => [base_icon_state]-[lightercolor] => lighter-r
+	var/lighter_color = null
 
-/obj/item/lighter/random/New()
-	..()
-	var/color = pick("r","c","y","g")
-	icon_state = "lighter-[color]"
+/obj/item/lighter/random/Initialize(mapload)
+	. = ..()
+	lighter_color = pick("r","c","y","g")
+	base_icon_state = "lighter"
```

Comment:
Why is this in `Initialize()` and not on the variable on the lighter itself

Reply:
Moved to `/obj/item/lighter/random`. Making this still work with zippos while also not cluttering the zippos.